backend/lib/endpoints/schema/location.js

```diff
@@ -0,0 +1,16 @@
+const S = require("fluent-schema");
+const { strictSchema } = require("./utils");
+
+const { LOCATION_TYPES } = require("../../models/Location");
+
+const locationSchema = strictSchema()
```

Comment:
Missing `state` and `zip`

Reply:
and how about required values? Can we have something like `oneOf`` at least one is required?

Reply:
As this is a shared "component" of a schema, I decided not to make any of them required. Can we compose those schema definitions? If not maybe I should make all of them required

Reply:
@Naraujo13 this is missing on the model also '-'

Reply:
Can we do this on another pull request? Location is a shared schema and I'm trying to merge users on this one. What do you think?

Reply:
The missing fields were added in the PR #403, they were really missing in the original model. 

About the required fields, I think this might be tricky here. I think what @robinv85 meant with the `oneOf` was that there is two options for the user to input the location:

- Only with coordinates, with no address - this way we can reverse geocode then to get full address
- Only the address (from gmaps list), with no coordinates - this way we geocode the coordinates to get full address 

This is to always keep coordinates and address consistent without needing to add extra checks in the backend.
So I think he meant a `oneOf` between those two options. 

Anyway, I wonder if this would be done by the backend or before sending the request to the API because if done before, the location would always be sent with all fields.